pkg/dyninst/procmon/analyze.go

```diff
@@ -296,3 +326,121 @@ func isGoElfBinaryWithDDTraceGo(f *os.File) (bool, error) {
 }
 
 var ddTraceSymbolSuffix = []byte("ddtrace/tracer.passProbeConfiguration")
+
+// executableAnalyzer is an interface for analyzing executables to determine if
+// they are interesting for dynamic instrumentation.
+type executableAnalyzer interface {
+	checkFileKeyCache(key FileKey) (interesting bool, known bool)
+	isInteresting(f *os.File, key FileKey) (bool, error)
+}
+
+// baseExecutableAnalyzer implements executableAnalyzer without any caching.
+type baseExecutableAnalyzer struct{}
+
+func (a *baseExecutableAnalyzer) checkFileKeyCache(
+	_ FileKey,
+) (interesting bool, known bool) {
+	return false, false
```

Comment:
If there's no cache, I think it'd make more sense to default to all processes being interesting.